internal/jet/literal_expression.go

```diff
@@ -326,6 +326,46 @@ func formatNanoseconds(nanoseconds ...time.Duration) string {
 	return ""
 }
 
+// ---------------------------------------------------//
+type rangeLiteral struct {
+	rangeInterfaceImpl
+	literalExpressionImpl
+}
+
+func RangeRaw(value string) RangeExpression {
+	rangeLiteral := rangeLiteral{}
+	rangeLiteral.literalExpressionImpl = *literal(value)
+	rangeLiteral.rangeInterfaceImpl.parent = &rangeLiteral
+
+	return &rangeLiteral
+}
+
+func DateRange(lowIncluded, highIncluded bool, lowDate, highDate time.Time) RangeExpression {
+	return RangeRaw(getRangeFromDate(lowIncluded, highIncluded, lowDate, highDate))
+}
+
+func getRangeFromDate(lowIncluded, highIncluded bool, lowDate, highDate time.Time) string {
```

Comment:
Postgres alrady defines how ranges are [constructed](https://www.postgresql.org/docs/16/rangetypes.html#RANGETYPES-CONSTRUCT). There is a third string parameter that defines if lower/upper bound is included.
```sql
SELECT numrange(1.0, 14.0, '(]');
```
To be consistent I think we should use the same convention.